platform/include/logging_stack.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Logging Stack V1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file logging_stack.h
+ * @brief Reference implementation of Logging stack as a header-only library.
+ */
+
+#ifndef LOGGING_STACK_
+#define LOGGING_STACK_
+
+/* Include header for logging level macros. */
+#include "logging_levels.h"
+
+/* Standard Include. */
+#include <stdio.h>
+
+/* Metadata information to prepend to every log message. */
+#define LOG_METADATA_FORMAT_ARGS    ( "[%s:%d] ", __FILE__, __LINE__ )
+
+/* Common macro for all logging interface macros. */
+#define SdkLog( string )               printf string
```

Comment:
Use of `printf` is not thread safe. Why are we not using the earlier implementation?

Reply:
I am looking to change this in a separate PR to avoid increasing the scope of this PR further. Within the implementation, I am also looking to support timestamp addition, thus, it may be slightly involved to support it only for a single call out of the 3 consecutive calls made per log message. I can add a TODO to mention about removing `printf`